src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/HostTests.cs

```diff
@@ -275,44 +275,53 @@ static string SaveRandomConfig(string appSettingsPath)
         [ActiveIssue("https://github.com/dotnet/runtime/issues/34582", TestPlatforms.Windows, TargetFrameworkMonikers.Netcoreapp, TestRuntimes.Mono)]
         public async Task CreateDefaultBuilder_SecretsDoesReload()
         {
-            var secretId = Assembly.GetExecutingAssembly().GetName().Name;
-            var reloadFlagConfig = new Dictionary<string, string>() { { "hostbuilder:reloadConfigOnChange", "true" } };
-            var secretPath = PathHelper.GetSecretsPathFromSecretsId(secretId);
-            var secretFileInfo = new FileInfo(secretPath);
-
-            Directory.CreateDirectory(secretFileInfo.Directory.FullName);
-
-            static string SaveRandomSecret(string secretPath)
+            var originalValue = Environment.GetEnvironmentVariable("DOTNET_USE_POLLING_FILE_WATCHER");
```

Comment:
Note for review: The diff looks tough to read but seems like this change is basically just adding a try/finally block around this entire code snippet.